internal/acceptance/git/git.go

```diff
@@ -42,13 +42,33 @@ import (
 
 type key int
 
-const (
-	gitHostKey         key = iota // we store the host:port under this in the Context
-	gitRepositoriesKey            // key to the path to the TEMP directory mounted in the git server container in the Context
-)
+const gitStateKey = key(0) // we store the gitState struct under this key in Context and when persisted
```

Comment:
For my own understanding, what's the point of using `key(0)` instead of `int(0)`, or simply `0`?

Reply:
When placed in `context.Context` we can end up with multiple keys having control over the same value, for example:

```go
const (
	keyA = 0
	keyB = 0
)

ctx := context.WithValue(context.TODO(), keyA, "A")
ctx = context.WithValue(context.TODO(), keyB, "B")

ctx.Value(keyA) == "B"
ctx.Value(keyB) == "B"
```

Giving each key a type, even though they have the same value makes them differ in the context; so these two do not control the same entry in Context:

```go

type keyA int
type keyB int

const (
	key1 = keyA(0)
	key2 = keyB(0)
)

ctx := context.WithValue(context.TODO(), keyA, "A")
ctx = context.WithValue(context.TODO(), keyA, "B")

ctx.Value(keyA) == "A"
ctx.Value(keyB) == "B"
```

(all examples written from the top of my head, so don't mind the syntax issues)

Reply:
This is great, thanks for taking the time to explain.